## Como leio e escrevo dados tabulares?

O pandas fornece a função read_csv(), para ler dados armazenados como um arquivo csv em um pandas DataFrame.  
O pandas suporta muitos formatos de arquivo ou fontes de dados diferentes prontos para uso (csv, excel, sql, json, parquet, …), cada um deles com o prefixo read_*.

In [12]:
import pandas as pd

Lendo o arquivo titanic.csv salvo na pasta data da raiz do projeto.  

Exemplo, do comando para ler uma planilha de um arquivo Excel  
titanic = pd.read_excel("../data/titanic.xlsx", sheet_name="passengers")

In [3]:
titanic = pd.read_csv("../data/titanic.csv")

Certifique-se de sempre verificar os dados após a leitura.  
Ao exibir um DataFrame, as primeiras e últimas 5 linhas serão exibidas por padrão:

In [5]:
titanic

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                                Heikkinen, Miss Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                        Graham, Miss Margaret Edith  female  19.0      0   
888            Johnston, Miss Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

Use o método heade(), para ver as primeiras linhas de um DataFrame ou o método .tail(), para ver as ultimas linhas.

Por padrão, o metodo head() e o tail(), exibe as 5 primeiras e ultimas linhas.  
Para exibir mas ou menos linhas, basta passar a quantidade como parâmetro.


In [8]:
titanic.head(3)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                              Heikkinen, Miss Laina  female  26.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S

In [9]:
titanic.tail(8)

PassengerId  Survived  Pclass                                     Name  \
883          884         0       2            Banfield, Mr. Frederick James   
884          885         0       3                   Sutehall, Mr. Henry Jr   
885          886         0       3     Rice, Mrs. William (Margaret Norton)   
886          887         0       2                    Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss Margaret Edith   
888          889         0       3  Johnston, Miss Catherine Helen "Carrie"   
889          890         1       1                    Behr, Mr. Karl Howell   
890          891         0       3                      Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch            Ticket    Fare Cabin Embarked  
883    male  28.0      0      0  C.A./SOTON 34068  10.500   NaN        S  
884    male  25.0      0      0   SOTON/OQ 392076   7.050   NaN        S  
885  female  39.0      0      5            382652  29.125   NaN        Q  
886    male  27.0      0      0            211536  13.000   NaN        S  
887  female  19.0      0      0            112053  30.000   B42        S  
888  female   NaN      1      2        W./C. 6607  23.450   NaN        S  
889    male  26.0      0      0            111369  30.000  C148        C  
890    male  32.0      0      0            370376   7.750   NaN        Q

Uma verificação de como o pandas interpretou cada um dos tipos de dados da coluna pode ser feita solicitando o atributo dtypes do pandas:

In [10]:
titanic.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

Para cada uma das colunas, o tipo de dado utilizado é listado.  
Os tipos de dado DataFramesão inteiros ( int64), flutuantes ( float64) e strings ( object).

Para obter um resumo técnico de um dataframe, use o método info()

In [11]:
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


O método info(), fornece informações técnicas sobre um DataFrame, explicando a saída com mais detalhes:

* É realmente um DataFrame.
* Há 891 entradas, ou seja, 891 linhas.
* Cada linha tem um rótulo de linha (também conhecido como index) com valores que variam de 0 a 890.
* A tabela tem 12 colunas. A maioria das colunas tem um valor para cada linha (todos os 891 valores são non-null). Algumas colunas têm valores ausentes e menos de 891 non-nullvalores.
* As colunas Name, Sexe consistem em dados textuais (strings, também conhecidas como ). As outras colunas são dados numéricos, alguns dos quais são números inteiros (também conhecidos como ) e outros são números reais (também conhecidos como Cabin) .Embarkedobjectintegerfloat
* O tipo de dados (caracteres, inteiros,…) nas diferentes colunas são resumidos listando o dtypes.
* A quantidade aproximada de RAM usada para armazenar o DataFrame também é fornecida.